tests/sles4sap/netweaver_ascs.pm

```diff
@@ -0,0 +1,68 @@
+# SUSE's SLES4SAP openQA tests
+
+# Summary: Checks NetWeaver's ASCS installation as performed by sles4sap/nw_ascs_install
+# Requires: sles4sap/nw_ascs_install
+# Maintainer: Alvaro Carvajal <acarvajal@suse.de>
+
+use base "x11test";
+use strict;
+use testapi;
+
+sub run {
+    my ($self) = @_;
+    my $pscmd = "ps auxw | grep ASCS | grep -vw grep";
+    $pscmd = "$pscmd | wc -l ; $pscmd";
+
+    x11_start_program('xterm');
+    assert_screen('xterm');
+
+    # The SAP Admin user 'qadadm' was set in sles4sap/nw_ascs_install
+    my $sapadmin = 'qadadm';
+
+    # Allow SAP Admin user to inform status via $testapi::serialdev
+    assert_script_sudo("chown $sapadmin /dev/$testapi::serialdev", 5);
+
+    type_string "su - $sapadmin\n";
+    assert_script_run("sapcontrol -nr 00 -function GetVersionInfo", 20);
+    assert_screen('netweaver-version-info', 10);
+
+    type_string "clear\n";
+    assert_script_run("sapcontrol -nr 00 -function GetInstanceProperties | grep ^SAP", 20);
+    assert_screen('netweaver-instance-properties', 10);
+
+    type_string "clear\n";
+    assert_script_run("sapcontrol -nr 00 -function Stop", 20);
+    assert_screen('netweaver-stop-instance', 10);
+
+    type_string "clear\n";
+    assert_script_run("sapcontrol -nr 00 -function StopService", 20);
+    assert_screen('netweaver-stop-service', 10);
+
+    type_string "clear\n";
+    type_string "$pscmd\n";
+    check_screen('netweaver-service-started', 30);
+
+    type_string "clear\n";
+    assert_script_run("sapcontrol -nr 00 -function StartService QAD", 20);
+    assert_screen('netweaver-start-service', 10);
+
+    type_string "clear\n";
+    type_string "$pscmd\n";
+    assert_screen('netweaver-service-started', 20);
+
+    type_string "clear\n";
+    assert_script_run("sapcontrol -nr 00 -function Start", 20);
+    assert_screen('netweaver-start-instance', 10);
+
+    type_string "clear\n";
+    type_string "$pscmd\n";
+    assert_screen('netweaver-instance-started', 20);
+
+    # Rollback changes to $testapi::serialdev and close the window
+    type_string "exit\n";
+    assert_script_sudo("chown $testapi::username /dev/$testapi::serialdev", 5);
```

Comment:
you just did exit from qaadm , so you are root currently . why you use sudo here ?